Mathlib/LinearAlgebra/Dimension/Free.lean

```diff
@@ -23,12 +23,55 @@ import Mathlib.LinearAlgebra.FreeModule.Finite.Basic
 
 noncomputable section
 
-universe u v v'
-
-variable {R : Type u} {M M₁ : Type v} {M' : Type v'}
+universe u v v' w
 
 open BigOperators Cardinal Basis Submodule Function Set DirectSum FiniteDimensional
 
+section Tower
+
+variable (F : Type u) (K : Type v) (A : Type w)
+
+variable [Ring F] [Ring K] [AddCommGroup A]
+
+variable [Module F K] [Module K A] [Module F A] [IsScalarTower F K A]
+
+variable [StrongRankCondition F] [StrongRankCondition K] [Module.Free F K] [Module.Free K A]
+
+/-- Tower law: if `A` is a `K`-module and `K` is an extension of `F` then
+$\operatorname{rank}_F(A) = \operatorname{rank}_F(K) * \operatorname{rank}_K(A)$. -/
```

Comment:
Can you add a reference to `rank_mul_rank` below?

Reply:
Like this?
```suggestion
$\operatorname{rank}_F(A) = \operatorname{rank}_F(K) * \operatorname{rank}_K(A)$.

The universe polymorphic version of `rank_mul_rank` below. -/
```